kernel/src/platform/mod.rs

```diff
@@ -7,26 +7,90 @@ pub mod mpu;
 crate mod systick;
 
 /// Interface for individual boards.
+///
+/// Each board in their `main.rs` file should define a struct which implements
+/// this trait. This trait is the core for how syscall dispatching is handled.
+///
+/// ## Example
+///
+/// ```ignore
+/// struct Hail {
+///     console: &'static capsules::console::Console<'static>,
+///     ipc: kernel::ipc::IPC,
+///     dac: &'static capsules::dac::Dac<'static>,
+/// }
+///
+/// impl Platform for Hail {
+///     fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
+///     where
+///         F: FnOnce(Option<&dyn kernel::Driver>) -> R,
+///     {
+///         match driver_num {
+///             capsules::console::DRIVER_NUM => f(Some(self.console)),
+///             kernel::ipc::DRIVER_NUM => f(Some(&self.ipc)),
+///             capsules::dac::DRIVER_NUM => f(Some(self.dac)),
+///
+///             _ => f(None),
+///         }
+///     }
+/// }
+/// ```
 pub trait Platform {
     /// Platform-specific mapping of syscall numbers to objects that implement
-    /// the Driver methods for that syscall
+    /// the Driver methods for that syscall.
     fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
     where
         F: FnOnce(Option<&dyn Driver>) -> R;
 }
 
 /// Interface for individual MCUs.
+///
+/// Each microcontroller should define a struct and implement this trait.
```

Comment:
"Defines chip-specific properties of Tock's operation. These include whether and which memory protection mechanism and systick to use, how to dispatch userland applications, and how to handle hardware events.